### PR TITLE
Feature manage tasktype

### LINKED ISSUE #10: Manage TaskType
## Expected behaviour
We want to be able to use a CRUD on `TaskType` objects.

## Current behavior
Actually we have no calls on the API
We do not have models too

## Potential solution
 - Create models
 - Create views (calls of the API)
 - Create a serializer for admin like : `AdminTaskTypeSerializer`

**1 - Create new TaskType :**
 - URL : `POST tasktypes/`
 - Restricted to `admin` only
 - params POST : `name`
 - return : the new tasktype serialization

**2 - Delete a tasktype :**
 - URL : `DELETE tasktypes/$id`
 - Restricted to `admin` only
 - params POST : no one
 - return : nothing

**3 - Update a tasktype :**
 - URL : `PATCH tasktypes/$id`
 - Restricted to `admin` only
 - params POST : `name`
 - return : the tasktype serialization (complete)

**4 - List tasktypes :**
 - URL : `GET tasktypes/`
 - Restricted to `admin` only
 - return : a list of all tasktypes serialization

**5 - Detail of a tasktype :**
 - URL : `GET tasktypes/$id`
 - Restricted to `admin` only
 - return : the tasktype serialization

### PR DESCRIPTION
**Pull request by DarkFranX**
| Q                                                      | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                      | Features
| Tickets (_issues_) concerned               | fixed #10 

---

##### What have you changed ?
- Create TaskType model 
- Add CRUD on TaskType objects
- Add tests for TaskType CRUD methods
- Fix minor bugs

##### How did you change it ?
- Add `models/`, `views/`, `serializers/` and `tests/` to the `volunteer/` app.
- Add `volunteer` app in `settings.py`
- Fix DEFAULT_FILTER_BACKENDS in `settings.py`
- Add an AdminFactory class in `apiVolontaria/factories.py`
- Add the `volunteer/` app URL in `apiVolontaria/urls.py`
- Register the `TaskType` model on the admin site in `volunteer/admin.py`
- Add a DB migration file to add the TaskType model in `volunteer/migrations/`
- Add tests for all methods in `volunteer/tests`

Verification :
 -  [X] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [X] This Pull-Request fully meets the requirements defined in the issue
 -  [X] I added or modified the attached tests

### CODE REVIEW

File: source/apiVolontaria/volunteer/models.py [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
 
-# Create your models here.
+
+class TaskType(models.Model):
+
+    """
+
+    This class represents the TaskType model.
+
+    """
```
Comment (RignonNoel) [inline source/apiVolontaria/volunteer/models.py]: Why do you put blank line in your comments all the time ?
It take a lot of lines and i never see that before

File: source/apiVolontaria/apiVolontaria/urls.py [hunk 2]
```diff
@@ -30,6 +30,11 @@
         ObtainTemporaryAuthToken.as_view(),
         name='token_api'
     ),
+    # Volunteer
+    url(
+        r'^volunteer/',
+        include('volunteer.urls', namespace="volunteer")
```
Comment (RignonNoel) [inline source/apiVolontaria/apiVolontaria/urls.py]: coma at end of the line

File: source/apiVolontaria/volunteer/urls.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from django.conf.urls import include, url
+from rest_framework.urlpatterns import format_suffix_patterns
+
+from . import views
+
+urlpatterns = format_suffix_patterns(
+    [
+        # Cycles
```
Comment (RignonNoel) [inline source/apiVolontaria/volunteer/urls.py]: It's not `Cycles` but `TaskTypes`

File: source/apiVolontaria/volunteer/models.py (hunk 1 above)
Comment (DarkFranX) [reply source/apiVolontaria/volunteer/models.py]: I took the idea from [this blog]() which summarized PEP257. Here is the concerned part:
> class Documentation(object):
>
>   """Illustrate class-level docstring.
>
>    Classes use a special whitespace convention: the opening and closing quotes
    are preceded and followed by a blank line, respectively. No other
    docstrings should be preceded or followed by anything but code.
>
>    A blank line at the end of a multi-line docstring before the closing
    quotation marks simply makes it easier for tooling to auto-format
    paragraphs (wrapping them at 79 characters, per PEP8), without the closing
    quotation marks interfering. For example, in Vim, you can use `gqip` to
    "apply text formatting inside the paragraph." In Emacs, the equivalent
    would be the `fill-paragraph` command. While it's not required, the
    presence of a blank line is quite common and much appreciated. Regardless,
    the closing quotation marks should certainly be on a line by themselves.
>
>    """

Maybe the newlines aren't really necessary in our case. We should establish a docstring convention of our own as soon as possible.

Comment (RignonNoel) [reply source/apiVolontaria/volunteer/models.py]: The link to the blog doesn't work (it's a link to this same page).

It's interessting, i've saw lot of people do some comment like this :

```
"""My first line
rest of
my content
"""
```

But to be more easy to manage with PEP8 i always wrote like this :

```
"""
My first line
rest of
my content
"""
```

Like it show in the [PEP257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) you quoted.

I've open a new issues on this point => #36

Comment (DarkFranX) [reply source/apiVolontaria/volunteer/models.py]: Great! Yep, generally the first line is a short sentence describing the content of the whole docstring.
> """Summary of the content
>
>Multiple line 
>of content
>
>"""

Thanks for the issue!

PS: Fixed the [link](http://blog.dolphm.com/pep257-good-python-docstrings-by-example/)

File: source/apiVolontaria/apiVolontaria/urls.py (hunk 2 above)
Comment (DarkFranX) [reply source/apiVolontaria/apiVolontaria/urls.py]: I'll fix the other url patterns then. I'll add a comma each

**Comment by RignonNoel** [comment]
Really good job :)

**Pull request merged**